agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalFileSender.java

```diff
@@ -42,36 +40,34 @@
 
   // send persisted telemetries from local disk every 30 seconds.
   private static final long INTERVAL_SECONDS = 30;
-  private static final ScheduledExecutorService scheduledExecutor =
-      Executors.newSingleThreadScheduledExecutor(
-          ThreadPoolUtils.createDaemonThreadFactory(LocalFileLoader.class));
 
   private final LocalFileLoader localFileLoader;
   private final TelemetryPipeline telemetryPipeline;
 
+  private final ScheduledExecutorService scheduledExecutor =
+      Executors.newSingleThreadScheduledExecutor(
+          ThreadPoolUtils.createDaemonThreadFactory(LocalFileLoader.class));
+
   private final TelemetryPipelineListener diagnosticListener =
       new DiagnosticTelemetryPipelineListener(
           "Sending telemetry to the ingestion service (retry from disk)");
 
-  static void start(LocalFileLoader localFileLoader, TelemetryPipeline telemetryPipeline) {
-    LocalFileSender localFileSender = new LocalFileSender(localFileLoader, telemetryPipeline);
+  LocalFileSender(LocalFileLoader localFileLoader, TelemetryPipeline telemetryPipeline) {
+    this.localFileLoader = localFileLoader;
+    this.telemetryPipeline = telemetryPipeline;
+
     scheduledExecutor.scheduleWithFixedDelay(
-        localFileSender, INTERVAL_SECONDS, INTERVAL_SECONDS, TimeUnit.SECONDS);
+        this, INTERVAL_SECONDS, INTERVAL_SECONDS, TimeUnit.SECONDS);
   }
 
-  private LocalFileSender(LocalFileLoader localFileLoader, TelemetryPipeline telemetryPipeline) {
-    this.localFileLoader = localFileLoader;
-    this.telemetryPipeline = telemetryPipeline;
+  void shutdown() {
+    scheduledExecutor.shutdown();
   }
 
   @Override
   public void run() {
     // TODO (heya) load all persisted files on disk in one or more batch per batch capacity?
     try {
-      if (Strings.isNullOrEmpty(TelemetryClient.getActive().getInstrumentationKey())) {
```

Comment:
this class can't depend on `TelemetryClient` anymore since it lives in azure-monitor-opentelemetry-exporter. I'm not sure if this check is needed, this is partly why I created https://msazure.visualstudio.com/One/_workitems/edit/13279991 to re-test lazy loading before 3.3.0 release

Reply:
i remembered now.. this is added to address azure spring cloud's feature ask.. to disable sending any telemetry when ikey is empty.  they used applicationinsights-rp.json to disable it. i think we need it.

Reply:
this is not sending new telemetry though, so is this condition really needed?

Reply:
we don't want to send telemetry from disk either when ikey is null.